### DISCUSSION TITLE
Does the library React-native-ranimated-2 require allowed newArchEnabled flag

### DISCUSSION THREAD

**Discussion by YOEL311** [software-mansion/react-native-reanimated#4045] [Q&A]
I use ``react-native-reanimated-2`` 

I understand that this library uses ``TurboModules`` and ``Hermes engine``

> Reanimated 2 is in an early version. As we wanted to share it with the community as soon as we could, the library still has some rough edges and limitations that we plan to address soon. Unfortunately some of the limitations come from the immaturity of React Native's **TurboModule** architecture that Reanimated 2 relies on. As a consequence we won't be able to support older versions of React Native and some issues that we yet plan to resolve may require full support of **TurboModules** which is not yet available to the public.

[source][1]

However, their documentation does not mention that you should enable it.

Further confusing is the fact that the value is false in the example project. 

It is explicitly written in the comment that it should be set to true in order to allow libraries to use it. 

````
# Use this property to enable support to the new architecture.
# This will allow you to use TurboModules and the Fabric render in
# your application. You should enable this flag either if you want
# to write custom TurboModules/Fabric components OR use libraries that
# are providing them.
newArchEnabled=false
````
[Official React-native-reanimated-2 example][2]

I am very confused. 

  [1]: https://docs.swmansion.com/react-native-reanimated/docs/next/#known-problems-and-limitations
  [2]: https://github.com/software-mansion/react-native-reanimated/blob/main/Example/android/gradle.properties
  
  _Originally posted by @YOEL311 in https://stackoverflow.com/q/75411094/9161478

**Comment by tomekzaw** [comment]
Hey @YOEL311, although Reanimated v2 uses JSI & TurboModules under the hood (no need to enable it manually), it doesn't support the New Architecture (in particular, the new React Native renderer, also known as Fabric), you need to upgrade to v3.

**Reply by YOEL311** [reply]
Thank you for your response

Can you write your response to my  StackOverflow question?

Alternatively, I could quote you there


https://stackoverflow.com/q/75411094/9161478

**Comment by YOEL311** [comment]
what about react-native-reanimated-3 
@tomekzaw 
Does my project have to be with the new architecture for use with the reanimated-3  version?

**Reply by tomekzaw** [reply]
@YOEL311 Nope, Reanimated v3 supports both Paper (current architecture) and Fabric (new architecture).
